batch/batch/server/database.py

```diff
@@ -64,12 +57,26 @@ class JobsTable(Table):
         await super().delete_record({'id': id})
 
     async def get_incomplete_parents(self, id):
-        parent_ids = await self.get_record(id, ['parent_ids'])
-        parent_ids = json.loads(parent_ids['parent_ids'])
-        parent_records = await self.get_records(parent_ids)
-        incomplete_parents = [pr['id'] for pr in parent_records.result()
-                              if pr['state'] == 'Created']
-        return incomplete_parents
+        async with self._db.pool.acquire() as conn:
+            async with conn.cursor() as cursor:
+                sql = f"""SELECT `id` FROM `{self.name}` WHERE `id` IN
+                          (SELECT `parent_id` FROM `{self._db.jobs_parents.name}` WHERE `job_id` = %s)
+                          AND `state` = %s""".replace('\n', ' ')
```

Comment:
It's a bit unusual to see query as a subquery rather than a join, but I suppose it doesn't much matter.
```
SELECT {self._db.jobs_parents.name}.parent_id
FROM {self.name}.id
INNER JOIN {self._db.jobs_parents.name} ON {self.name}.id = {self._db.jobs_parents.name}.id
WHERE {self.name}.state = %s AND {self._db.jobs_parents.name}.job_id = %s
```